samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDialog.java

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.core;
+
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.dialogs.DialogTurnResult;
+import com.microsoft.bot.dialogs.WaterfallDialog;
+import com.microsoft.bot.dialogs.WaterfallStep;
+import com.microsoft.bot.dialogs.WaterfallStepContext;
+import com.microsoft.bot.dialogs.prompts.ConfirmPrompt;
+import com.microsoft.bot.dialogs.prompts.PromptOptions;
+import com.microsoft.bot.dialogs.prompts.TextPrompt;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.InputHints;
+import com.microsoft.recognizers.datatypes.timex.expression.Constants;
+import com.microsoft.recognizers.datatypes.timex.expression.TimexProperty;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ *
+ */
+public class BookingDialog extends CancelAndHelpDialog {
+    private final String destinationStepMsgText = "Where would you like to travel to?";
+    private final String originStepMsgText = "Where are you traveling from?";
+
+    /**
+     *
+     */
+    public BookingDialog() {
+        super(BookingDialog.class.getName());
+
+        addDialog(new TextPrompt(TextPrompt.class.getName()));
+        addDialog(new ConfirmPrompt(ConfirmPrompt.class.getName()));
+        addDialog(new DateResolverDialog(null));
+        WaterfallStep[] waterfallSteps = {
+            this::destinationStep,
+            this::originStep,
+            this::travelDateStep,
+            this::confirmStep,
+            this::finalStep
+        };
+        addDialog(new WaterfallDialog(WaterfallDialog.class.getName(), Arrays.asList(waterfallSteps)));
+
+        // The initial child Dialog to run.
+        setInitialDialogId(WaterfallDialog.class.getName());
+    }
+
+    private CompletableFuture<DialogTurnResult> destinationStep(WaterfallStepContext stepContext) {
+        BookingDetails bookingDetails = (BookingDetails) stepContext.getOptions();
+
+        if (bookingDetails.getDestination() == null) {
+            Activity promptMessage =
+                MessageFactory.text(destinationStepMsgText, destinationStepMsgText, InputHints.EXPECTING_INPUT);
+            return stepContext.prompt(TextPrompt.class.getName(), new PromptOptions() {{
+                setPrompt(promptMessage);
+            }});
+        }
+
+        return stepContext.next(bookingDetails.getDestination());
+    }
+
+    private CompletableFuture<DialogTurnResult> originStep(WaterfallStepContext stepContext) {
+        BookingDetails bookingDetails = (BookingDetails) stepContext.getOptions();
+
+        bookingDetails.setDestination(stepContext.getResult().toString());
+
+        if (StringUtils.isNotBlank(bookingDetails.getOrigin())) {
```

Comment:
```suggestion
        if (bookingDetails.getOrigin() == null) {
```

[C#](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/13.core-bot/Dialogs/BookingDialog.cs#L56) checks if it's null

Reply:
I've been looking at these comments. I've doubts -at least in [C#](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/13.core-bot/Dialogs/BookingDialog.cs#L56)- that string can ever be null, unless it's a class parsed and properties are generated. I'll change it to null anyway in our Java code, and we can later check the bot running

Reply:
Ack, however, you added `isNotBlank` and it should be `isBlank` in this case. `isBlank` check if it's null or an empty String

Reply:
yes, the check was reversed, I realized now